lib/binding/kvdb_interface.c

```diff
@@ -130,37 +130,39 @@ rest_get_dt(
     enum rest_status status = REST_STATUS_OK;
 
     err = rest_params_get(req->rr_params, "pretty", &pretty, false);
-    if (err)
-        return REST_STATUS_BAD_REQUEST;
+    if (ev(err))
+        return rest_response_perror(resp, REST_STATUS_BAD_REQUEST,
+            "The 'pretty' query parameter must be a boolean");
```

Comment:
We should prob get rid of the "pretty" option, piping the output into jq does an awesome job and is configurable...

Reply:
Is that Greg?

Reply:
You bring up a good point.

Reply:
The thought has occurred to me that we could get rid of all output format options and use go templates.  Github CLI supports this.  See `gh pr list --template`  and `gh help formatting`.  I have a template that produces this output using `gh pr list`:
```
PR #530    Support varying value lengths from run to run

   APROVED: alexttx

   merge:  MERGEABLE
   url:    https://github.com/hse-project/hse/pull/530
   author: beckerg


PR #528    Fix HSE_ERR_CTX_MAX

   APROVED: alexttx
   APROVED: beckerg

   merge:  MERGEABLE
   url:    https://github.com/hse-project/hse/pull/528
   author: tristan957

   checks:                 codeql
   checks:                 normal (fedora-36, gcc, release)
   checks:                 normal (fedora-36, gcc, debug)
   checks:                 normal (fedora-36, clang, release)
   checks:                 normal (fedora-36, clang, debug)
   checks:                 normal (ubuntu-18.04, gcc, release)
   checks:                 normal (ubuntu-18.04, gcc, debug)
   checks:                 normal (ubuntu-18.04, clang, release)
   checks:                 normal (ubuntu-18.04, clang, debug)
   checks:                 normal (ubuntu-22.04, gcc, release)
   checks:                 normal (ubuntu-22.04, gcc, debug)
   checks:                 normal (ubuntu-22.04, clang, release)
   checks:                 normal (ubuntu-22.04, clang, debug)
   checks:                 asan-ubsan (release)
   checks:                 asan-ubsan (debug)
   checks:                 lto (release)
   checks:                 lto (debug)
...
```

Reply:
Could you expand on this a bit? Are you suggesting this tool could be written in Go with pre-formatted templates and the addition to have your own?

Reply:
No.  I assume there are libraries that will do the templating, in which case `hsettp` could support something like this:
```
hsettp ... --template csched.template
```
Or maybe `jq`-like tool that will process a JSON document with a Go template:
```
hsettp ... | jqliketool --template csched.template
```

Reply:
If you find one, let me know. Although, I think rewriting the CLI in Go has other benefits too.

Reply:
Interviewing at google?

Reply:
better tooling for generating good CLIs